errors.go

```diff
@@ -0,0 +1,12 @@
+package gapi
+
+import "fmt"
+
+type ErrNotFound struct {
+	StatusCode   int
+	BodyContents []byte
+}
+
+func (e ErrNotFound) Error() string {
+	return fmt.Sprintf("status: %d, body: %v", e.StatusCode, string(e.BodyContents))
```

Comment:
This could be simplified:
```suggestion
	return fmt.Sprintf("status: %d, body: %s", e.StatusCode, e.BodyContents)
```

Which when combined with my other comment, would end up being:

```suggestion
	return fmt.Sprintf("status: 404 Not Found, body: %s", e.BodyContents)
```

I honestly would even remove the `status: 404` prefix, however, that would break backwards compatibility with consumers of this package that are looking for the prefix (like [here](https://github.com/grafana/terraform-provider-grafana/blob/8db6565d8826a709f12c4d975926960273f04d90/internal/resources/grafana/resource_alerting_contact_point.go#L114)).

Reply:
Thanks @inkel, I applied your feedback at https://github.com/grafana/grafana-api-golang-client/pull/163/commits/110dc0411315aff657ca2bcb6e573491ce682391, because it's indeed a nice idea! 

However, I think in your suggestion there was an extra `"Not Found"`, cause the former error was just printing including the status code (integer).

Could you confirm it looks good now, please? Thanks! 🙏🏻

Reply:
Nice, thanks! Yeah, the extra _not found_ was there on purpose, so in case the user prints the error message it shows indeed that it's not found (`status: 404` says the same, but you require some technical knowledge); but I'm ok without having it in the message for now. Re-approving and merging!